### DISCUSSION TITLE
Broken AppImage

### DISCUSSION THREAD

**Discussion by 7deok7** [ppy/osu#30466] [Q&A]
### Type

Other

### Bug description

When you download the .appimage file from the osu! website and run it through the AppImage Launcher (An app used to integrate appimage applications instead of them being a file on their own) you get an error window with the text: "Failed to register AppImage in system via libappimage"
This is not AppImage Launcher's fault, but rather the appimage itself being broken in some way, as said in this comment on github: https://github.com/TheAssassin/AppImageLauncher/issues/410#issuecomment-830989373
The comment also mentions how to look for the problem in the appimage, and that is by using appimagelint. This is the output when I run the appimage through it:

`appimagelint.cli[4397] [INFO] Checking AppImage ./osu.AppImage
appimagelint.cli[4397] [INFO] Running check "GNU libc ABI check"
appimagelint.glibc_abi_check[4397] [INFO] detected required version for runtime: 2.14
appimagelint.glibc_abi_check[4397] [INFO] detected required version for payload: 2.29
appimagelint.cache[4397] [INFO] Fetching glibc version data for debian
appimagelint.setup[4397] [INFO] Fetching glibc package versions from Debian sources API
appimagelint.glibc_abi_check[4397] [INFO] [✔] AppImage can run on Debian oldstable (bullseye)
appimagelint.glibc_abi_check[4397] [INFO] [✔] AppImage can run on Debian stable (bookworm)
appimagelint.glibc_abi_check[4397] [INFO] [✔] AppImage can run on Debian testing (trixie)
appimagelint.glibc_abi_check[4397] [INFO] [✔] AppImage can run on Debian unstable (sid)
appimagelint.cache[4397] [INFO] Fetching glibc version data for ubuntu
appimagelint.setup[4397] [INFO] Fetching glibc package versions from Ubuntu FTP mirror
appimagelint.glibc_abi_check[4397] [INFO] [✔] AppImage can run on Ubuntu oracular
appimagelint.glibc_abi_check[4397] [INFO] [✔] AppImage can run on Ubuntu noble
appimagelint.glibc_abi_check[4397] [INFO] [✔] AppImage can run on Ubuntu jammy
appimagelint.glibc_abi_check[4397] [INFO] [✔] AppImage can run on Ubuntu focal
appimagelint.glibc_abi_check[4397] [INFO] [✖] AppImage can run on Ubuntu bionic
appimagelint.glibc_abi_check[4397] [INFO] [✖] AppImage can run on Ubuntu xenial
appimagelint.glibc_abi_check[4397] [INFO] [✖] AppImage can run on Ubuntu trusty
appimagelint.cache[4397] [INFO] Fetching glibc version data for centos
appimagelint.setup[4397] [INFO] Fetching glibc package versions from CentOS mirror
appimagelint.cache[4397] [WARNING] codebase changed since last update, but updating failed, using cached data
appimagelint.cache[4397] [ERROR] 410 Client Error: Gone for url: https://ftp.fau.de/centos/7/os/x86_64/repodata//repomd.xml
Traceback (most recent call last):
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/json_cache_impl_base.py", line 101, in get_data
    new_data = cls._fetch_data()
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/package_version_maps.py", line 17, in _fetch_data
    return get_map_callback()
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/common.py", line 252, in get_centos_glibc_versions_map
    return get_centos_package_versions_map("glibc", r"libc\.so\.6\(GLIBC_(.*)\)")
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/common.py", line 151, in get_centos_package_versions_map
    primary_xml_root = ET.fromstring(get_centos_repodata_primary_xml(release))
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/common.py", line 126, in get_centos_repodata_primary_xml
    repomd_xml_root = ET.fromstring(get_request(f"{base_url}/repomd.xml").content)
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/common.py", line 119, in get_request
    response.raise_for_status()
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 410 Client Error: Gone for url: https://ftp.fau.de/centos/7/os/x86_64/repodata//repomd.xml
appimagelint.glibc_abi_check[4397] [INFO] [✖] AppImage can run on CentOS 7
appimagelint.cli[4397] [INFO] Running check "GNU libstdc++ ABI check"
appimagelint.glibcxx_abi_check[4397] [INFO] detected required version for runtime: <none>
appimagelint.glibcxx_abi_check[4397] [INFO] detected required version for payload: 3.4.26
appimagelint.cache[4397] [INFO] Fetching glibcxx version data for debian
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-c9zzel1s.tmp/out/
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-_cicsfdt.tmp/out/
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-hx2uv3gw.tmp/out/
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-36zltzva.tmp/out/
appimagelint.glibcxx_abi_check[4397] [INFO] [✔] AppImage can run on Debian oldstable (bullseye)
appimagelint.glibcxx_abi_check[4397] [INFO] [✔] AppImage can run on Debian stable (bookworm)
appimagelint.glibcxx_abi_check[4397] [INFO] [✔] AppImage can run on Debian testing (trixie)
appimagelint.glibcxx_abi_check[4397] [INFO] [✔] AppImage can run on Debian unstable (sid)
appimagelint.cache[4397] [INFO] Fetching glibcxx version data for ubuntu
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-vv6b3zul.tmp/out/
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-f3jja0tw.tmp/out/
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-ntwfp_vm.tmp/out/
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-wccd9ln2.tmp/out/
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-qeipyfwx.tmp/out/
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-_b_rqjp2.tmp/out/
appimagelint.gnu_lib_versions_symbols_finder[4397] [WARNING] no binaries found in /dev/shm/appimagelint-5_z8h4st.tmp/out/
appimagelint.glibcxx_abi_check[4397] [INFO] [✔] AppImage can run on Ubuntu oracular
appimagelint.glibcxx_abi_check[4397] [INFO] [✔] AppImage can run on Ubuntu noble
appimagelint.glibcxx_abi_check[4397] [INFO] [✔] AppImage can run on Ubuntu jammy
appimagelint.glibcxx_abi_check[4397] [INFO] [✔] AppImage can run on Ubuntu focal
appimagelint.glibcxx_abi_check[4397] [INFO] [✖] AppImage can run on Ubuntu bionic
appimagelint.glibcxx_abi_check[4397] [INFO] [✖] AppImage can run on Ubuntu xenial
appimagelint.glibcxx_abi_check[4397] [INFO] [✖] AppImage can run on Ubuntu trusty
appimagelint.cache[4397] [INFO] Fetching glibcxx version data for centos
appimagelint.setup[4397] [INFO] Fetching libstdc++ package versions from CentOS mirror
appimagelint.cache[4397] [WARNING] codebase changed since last update, but updating failed, using cached data
appimagelint.cache[4397] [ERROR] 410 Client Error: Gone for url: https://ftp.fau.de/centos/7/os/x86_64/repodata//repomd.xml
Traceback (most recent call last):
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/json_cache_impl_base.py", line 101, in get_data
    new_data = cls._fetch_data()
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/package_version_maps.py", line 17, in _fetch_data
    return get_map_callback()
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/common.py", line 256, in get_centos_glibcxx_versions_map
    return get_centos_package_versions_map("libstdc++", r"libstdc\+\+\.so\.6\(GLIBCXX_(.*)\)")
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/common.py", line 151, in get_centos_package_versions_map
    primary_xml_root = ET.fromstring(get_centos_repodata_primary_xml(release))
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/common.py", line 126, in get_centos_repodata_primary_xml
    repomd_xml_root = ET.fromstring(get_request(f"{base_url}/repomd.xml").content)
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cache/common.py", line 119, in get_request
    response.raise_for_status()
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/requests/models.py", line 943, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 410 Client Error: Gone for url: https://ftp.fau.de/centos/7/os/x86_64/repodata//repomd.xml
appimagelint.glibcxx_abi_check[4397] [INFO] [✖] AppImage can run on CentOS 7
appimagelint.cli[4397] [INFO] Running check "Icons validity and location check"
appimagelint.icons_check[4397] [INFO] Extracting icon name from desktop file: /tmp/.mount_osu.ApxLzii3/osu!.desktop
appimagelint.icons_check[4397] [INFO] Checking resolution of icon: /tmp/.mount_osu.ApxLzii3/osu!.png
appimagelint.icons_check[4397] [WARNING] icon X resolution 1024 is unknown, icon will most likely not be used and just wastes space
appimagelint.icons_check[4397] [WARNING] icon Y resolution 1024 is unknown, icon will most likely not be used and just wastes space
appimagelint.icons_check[4397] [INFO] [✔] Valid icon in AppDir root
appimagelint.icons_check[4397] [INFO] Checking resolution of icon: /tmp/.mount_osu.ApxLzii3/.DirIcon
Traceback (most recent call last):
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/__main__.py", line 4, in <module>
    run()
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/cli.py", line 142, in run
    for testres in check.run():
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/checks/icons.py", line 106, in run
    dotdiricon_valid = self._check_icon_for_valid_resolution(op.join(mountpoint, ".DirIcon"))
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/checks/icons.py", line 279, in _check_icon_for_valid_resolution
    res = self._get_icon_res(icon_path)
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/checks/icons.py", line 245, in _get_icon_res
    is_svg = self._is_svg(icon_path)
  File "/tmp/.mount_appima4zu5qJ/usr/conda/lib/python3.9/site-packages/appimagelint/checks/icons.py", line 222, in _is_svg
    with open(icon_path) as f:
FileNotFoundError: [Errno 2] No such file or directory: '/tmp/.mount_osu.ApxLzii3/.DirIcon'
`
Yes, it's true that you can integrate osu!lazer from flatpak, but the issue with flatpak is that it breaks a lot of the functionality of the client (accessing the client's folders for example, like exports, external beatmap editing, the osu! folder itself, etc.)

The game still runs fine (besides the fact that the icon on the application is gone), it just might lead to other issues in the future.

### Screenshots or videos

![image](https://github.com/user-attachments/assets/c06acbfb-9359-4038-9b06-ef8e085d9c20)

### Version

2024.1009.1

### Logs

[compressed-logs.zip](https://github.com/user-attachments/files/17589003/compressed-logs.zip)

**Comment by smoogipoo** [comment]
Dunno. We don't support running through AIL and have done our best to support it (rolling back certain upgrades in order to do so) even though it is now deprecated software that _hasn't been updated in over a year_.

You're unlikely to see a resolution here, sorry.

**Reply by 7deok7** [reply]
I see, thanks

**Reply by ravener** [reply]
I found the root issue of this, check out the notes I left on https://github.com/ppy/osu/issues/30759
